tests/run_tests.py

```diff
@@ -47,37 +78,55 @@
     # Per https://docs.python.org/2/library/argparse.html#partial-parsing, gather any known args. These will be passed to pytest.
     parsed_args, extra_args = parser.parse_known_args()
 
-    if parsed_args.rebuildgrades:
-        with pushd('../../..'):
-            print("recalculating grades tables")
-            xqt('{} web2py.py -S runestone -M -R applications/runestone/tests/make_clean_db_with_grades.py'.format(sys.executable))
-            print("dumping the data")
-            xqt('pg_dump --no-owner runestone_test > applications/runestone/tests/runestone_test.sql')
-        sys.exit(0)
-
     if parsed_args.skipdbinit:
         print('Skipping DB initialization.')
     else:
-        # make sure runestone_test is nice and clean.
-        xqt('dropdb --echo --if-exists --host={} --user={} "{}"'.format(pgnetloc, pguser, dbname),
-            'createdb --echo --host={} --user={} "{}"'.format(pgnetloc, pguser, dbname),
-            'psql  --host={} --user={} "{}" < runestone_test.sql'.format(pgnetloc, pguser, dbname))
-        # Copy the test book to the books directory.
-        rmtree('../books/test_course_1', ignore_errors=True)
-        # Sometimes this fails for no good reason on Windows. Retry.
-        for retry in range(100):
-            try:
-                copytree('test_course_1', '../books/test_course_1')
-                break
-            except WindowsError:
-                if retry == 99:
-                    raise
-        # Build the test book to add in db fields needed.
-        with pushd('../books/test_course_1'):
-            # The runestone build process only looks at ``DBURL``.
-            os.environ['DBURL'] = os.environ['TEST_DBURL']
-            xqt('{} -m runestone build --all'.format(sys.executable),
-                '{} -m runestone deploy'.format(sys.executable))
+        # Make sure runestone_test is nice and clean -- this will remove many
+        # tables that web2py will then re-create.
+        xqt('dropdb --echo --if-exists --host={} --username={} "{}"'.format(pgnetloc, pguser, dbname),
+            'createdb --echo --host={} --username={} "{}"'.format(pgnetloc, pguser, dbname),
+            'psql  --host={} --username={} "{}" < runestone_test.sql'.format(pgnetloc, pguser, dbname))
+        # Tell web2py that tables have been removed from the database so that
+        # its migration algorithm will re-create them.
+        remove_web2py_tables()
+
+        if parsed_args.rebuildgrades:
+            # Paranoia: if the existing ``runestone_test.sql`` database contains
+            # broken tables, remove all tables that web2py can re-create to
+            # return the databased to a fixed state.
+            xqt('psql "{}" < runestone_clean.sql'.format(dbname))
```

Comment:
This needs to use the host and username information like the other psql commands.  We cannot assume that the database will always be running on the same host as the tests, and in fact in our Docker configuration the database is in a separate container.

Reply:
Good point -- thanks. I'll update my PR. I'm still seeing some strange Docker things, so I'm rebuilding my VM / Docker setup and hope to debug this soon.